skopt/sample.py

```diff
@@ -0,0 +1,75 @@
+from collections import Mapping
+from numbers import Integral
+from numbers import Real
+
+from scipy.stats._distn_infrastructure import rv_frozen
+from scipy.stats.distributions import randint
+from scipy.stats.distributions import uniform
+
+from sklearn.utils import check_random_state
+from sklearn.utils.fixes import sp_version
+
+
+class Categorical:
+    def __init__(self, *args):
+        self.values = args
+        self.low = min(self.values)
+        self.high = max(self.values)
+        self._rvs = randint(0, len(self.values))
+
+    def rvs(self, random_state=None):
+        return self.values[self._rvs.rvs(random_state=random_state)]
+
+
+class Discrete:
+    def __init__(self, low, high):
+        self.low = low
+        self.high = high
+        self._rvs = randint(self.low, self.high)
+
+    def rvs(self, random_state=None):
+        return self._rvs.rvs(random_state=random_state)
+
+
+class Continuous:
```

Comment:
For me `Uniform` doesn't convey the idea that it is about continuous values :-/ You could have a uniform random variable over integers as well.